wp-cli/vip-filesystem.php

```diff
@@ -0,0 +1,177 @@
+	 * : Wether or not to update to database, or simply inspect it.
+	 * ---
+	 * default: false
+	 * options:
+	 *   - true
+	 *   - false
+	 * ---
+	 *
+	 * [--batch=<batch>]
+	 * : Batch size to process attachments in.
+	 * ---
+	 * default: 1000
+	 * ---
+	 *
+	 * ## EXAMPLES
+	 *     wp vip files update-filesizes
+	 *
+	 * @subcommand update-filesizes
+	 * @sypnosis [--dry-run=<dry-run>] [--batch=<batch>]
+	 */
+	public function update_filesizes( $args, $assoc_args ) {
+		global $wpdb;
+
+		$offset = 0;
+
+		WP_CLI::line( 'Updating attachment filesize metadata...' );
+
+		$log_file_name = sprintf( '%svip-files-update-filesizes-%s%s.csv', get_temp_dir(), $this->dry_run ? 'dry-run-' : '', date( 'YmdHi' ) );
+		$this->log_file = fopen( $log_file_name, 'w' );
+
+		// Parse arguments
+		$_dry_run = WP_CLI\Utils\get_flag_value( $assoc_args, 'dry-run', true );
+		if ( 'false' === $_dry_run ) {
+			$this->dry_run = false;
+		}
+
+		$batch_size = (int) WP_CLI\Utils\get_flag_value( $assoc_args, 'batch', 1000 );
+		if ( 0 >= $batch_size ) {
+			WP_CLI::error( 'Invalid batch size: ' . $batch_size );
+			WP_CLI::halt( 1 );
+		}
+
+		WP_CLI::line( '' );
+		WP_CLI::line( 'ARGUMENTS' );
+		WP_CLI::line( '* dry run: ' . ( $this->dry_run ? 'yes' : 'no' ) );
+		WP_CLI::line( '* batch size: ' . $batch_size );
+		WP_CLI::line( '' );
+
+		$attachment_count = array_sum( (array) wp_count_posts( 'attachment' ) );
+		if ( 0 >= $attachment_count ) {
+			WP_CLI::error( 'No attachments found' );
+			WP_CLI::halt( 1 );
+		}
+
+		$this->progress = \WP_CLI\Utils\make_progress_bar(
+			'Checking ' . number_format( $attachment_count ) . ' attachments', $attachment_count );
+
+		do {
+			$sql = $wpdb->prepare( 'SELECT ID FROM ' . $wpdb->posts . ' WHERE post_type = "attachment" LIMIT %d, %d',
+				$offset, $batch_size );
+			$attachments = $wpdb->get_results( $sql );
+
+			if ( $attachments ) {
+				$this->update_attachments( $attachments );
+			}
+
+			// Pause.
+			$this->stop_the_insanity();
+			sleep( 1 );
+
+			$offset += $batch_size;
+
+		} while ( count( $attachments ) );
+
+		$this->progress->finish();
+
+		fclose( $this->log_file );
+
+		WP_CLI::success( 'Attachments metadata update complete!' );
+		WP_CLI::success( 'Log file can be found at: ' . $log_file_name );
+	}
+
+	/**
+	 * Update attachments' metadata
+	 *
+	 * @param array $attachments
+	 */
+	private function update_attachments( array $attachments ): void {
+		foreach ( $attachments as $attachment ) {
+			list( $did_update, $result ) = $this->update_attachment_filesize( $attachment->ID );
+
+			fputcsv( $this->log_file, [
+				$attachment->ID,
+				$did_update ? 'updated' : 'skipped',
+				$result,
+			] );
+		}
+	}
+
+	/**
+	 * Update attachment's filesize metadata
+	 *
+	 * @param int $attachment_id
+	 */
+	private function update_attachment_filesize( $attachment_id ): array {
+		$this->progress->tick();
+
+		$meta = wp_get_attachment_metadata( $attachment_id );
+
+		if ( ! is_array( $meta ) ) {
```

Comment:
Would simply checking for `false` be inadequate here? Since the possible values are either `false` or the `_wp_attachment_metadata` array.

Reply:
I've come across cases before where the value is neither and we end up with warnings about undefined indexes (and then corrupted data saved).